src/context.ts

```diff
@@ -783,34 +700,24 @@ export class Context<
     arg?: U | ResponseOrInit<U>,
     headers?: HeaderRecord
   ): JSONRespondReturn<T, U> => {
-    const body = JSON.stringify(object)
-    this.#preparedHeaders ??= {}
-    this.#preparedHeaders['content-type'] = 'application/json'
-    /* eslint-disable @typescript-eslint/no-explicit-any */
-    return (
-      typeof arg === 'number' ? this.#newResponse(body, arg, headers) : this.#newResponse(body, arg)
-    ) as any
+    return this.#newResponse(
+      JSON.stringify(object),
+      arg,
+      setDefaultContentType('application/json', headers)
+    ) /* eslint-disable @typescript-eslint/no-explicit-any */ as any
   }
 
   html: HTMLRespond = (
     html: string | Promise<string>,
     arg?: ContentfulStatusCode | ResponseOrInit<ContentfulStatusCode>,
     headers?: HeaderRecord
   ): Response | Promise<Response> => {
-    this.#preparedHeaders ??= {}
-    this.#preparedHeaders['content-type'] = 'text/html; charset=UTF-8'
-
-    if (typeof html === 'object') {
-      return resolveCallback(html, HtmlEscapedCallbackPhase.Stringify, false, {}).then((html) => {
-        return typeof arg === 'number'
-          ? this.#newResponse(html, arg, headers)
-          : this.#newResponse(html, arg)
-      })
-    }
-
-    return typeof arg === 'number'
-      ? this.#newResponse(html as string, arg, headers)
-      : this.#newResponse(html as string, arg)
+    const TEXT_HTML = 'text/html; charset=UTF-8'
+    const res = (html: string) =>
+      this.#newResponse(html, arg, setDefaultContentType(TEXT_HTML, headers))
```

Comment:
I think it's okay to specify it directly, but do you need to put it in a variable first?

```suggestion
      this.#newResponse(html, arg, setDefaultContentType('text/html; charset=UTF-8', headers))
```

Reply:
You are right! Fixed https://github.com/honojs/hono/pull/4100/commits/47bb3287b7296363086135ec15b13a654c5fb387